src/de.rs

```diff
@@ -311,6 +311,12 @@ impl<'de> de::Deserializer<'de> for Deserializer {
             visitor.visit_string(v)
         } else if Array::is_array(&self.value) {
             self.deserialize_seq(visitor)
+        } else if self.value.is_instance_of::<Uint8Array>()
+            || self.value.is_instance_of::<ArrayBuffer>()
+        {
+            // We need to handle this here because serde uses `deserialize_any`
+            // for internally tagged enums
+            self.deserialize_byte_buf(visitor)
```

Comment:
I will add another variant to the enum test to make sure it round trips safely without the attribute - but in manual testing, if you don't have `#[serde(with = "serde_bytes)]` you will get an `Array<number>` on the JS side.

Reply:
> but in manual testing, if you don't have #[serde(with = "serde_bytes)] you will get an Array<number> on the JS side.

Oh yeah but this is changing deserialization, so the question is about opposite direction - I'm wondering what happens when you convert from Uint8Array to a Rust enum that doesn't have serde_bytes but regular Vec<u8>.

Reply:
In that case I get the error `Error: invalid type: byte array, expected a sequence`.

Reply:
That seems unfortunate and matches a comment I left a while back a bit further below (that bytes buf is expected only in response to explicit deseriliaze_buf).

I wonder if deserializing those types as a sequence would make sense so that it would cover enums with regular vecs too... At least it would make sense for Uint8Array, probably not so much for ArrayBuffer.

Reply:
OTOH for now this would be a strictly additive change, and current PR's implementation is faster, so let's go with it for now.

Reply:
Yea, I am sure a cleaner approach can be found that fixes having to add more cases here - but in the short term this seems to work well (I am using it in an in progress [SQLSync](https://github.com/orbitinghail/sqlsync) PR)